sdk/Lusid.Sdk.Tests/tutorials/Instruments/DemoFxOption.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Lusid.Sdk.Model;
+using Lusid.Sdk.Tests.Utilities;
+using Lusid.Sdk.Utilities;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace Lusid.Sdk.Tests.Tutorials.Instruments
+{
+    [TestFixture]
+    public class DemoFxOption: DemoInstrumentBase
+    {
+        internal override void CreateMarketData(string scope, ModelSelection.ModelEnum model, LusidInstrument fxOption)
+        {
+            // POPULATE with required market data for valuation of the instruments
+            var upsertFxRateRequestreq = TestDataUtilities.BuildFxRateRequest(scope, TestDataUtilities.EffectiveAt);
+            var upsertQuoteResponse = _quotesApi.UpsertQuotes(scope, upsertFxRateRequestreq);
+            
+            ValidateQuoteUpsert(upsertQuoteResponse, upsertFxRateRequestreq.Count);
+
+            List<Dictionary<string, UpsertComplexMarketDataRequest>> complexMarket =
+                new List<Dictionary<string, UpsertComplexMarketDataRequest>>();
+            if (model != ModelSelection.ModelEnum.ConstantTimeValueOfMoney)
```

Comment:
Maybe add some comments describing the relationship between the model and the type of market data it requires